src/github.js

```diff
@@ -14,62 +14,65 @@
  * limitations under the License.
  */
 
-const logging = require('./logging').default;
 const {RepoFile} = require('./repo-file');
-const config = require('../config');
-const bb = require('bluebird');
+const {Owner} = require('./owner');
+const {Git} = require('./git');
 const _ = require('lodash');
 
-const request = bb.promisify(require('request'));
-const GITHUB_ACCESS_TOKEN = config.get('GITHUB_ACCESS_TOKEN');
-
-const headers = {
-  'User-Agent': 'get-owners',
-  'Accept': 'application/vnd.github.v3+json',
-};
-const qs = {
-  access_token: GITHUB_ACCESS_TOKEN,
-};
 
 /**
  * Maps the github json payload to a simpler data structure.
  */
-export class PullRequest {
+class PullRequest {
 
-  constructor(json) {
-    this.id = json.number;
-    this.author = json.user.login;
-    this.state = json.state;
+  constructor(context, pr) {
 
-    // Ref here is the branch name
-    this.headRef = json.head.ref;
-    this.headSha = json.head.sha;
+    this.git = new Git(context);
+    this.context = context;
+    this.github = context.github;
+
+    this.id = pr.number;
+    this.author = pr.user.login;
+    this.state = pr.state;
 
-    this.baseRef = json.base.ref;
-    this.baseSha = json.base.sha;
+    this.owner = pr.base.repo.owner.login;
+    this.repo = pr.base.repo.name;
 
-    this.project = json.base.repo.owner.login;
-    this.repo = json.base.repo.name;
-    this.cloneUrl = json.base.repo.clone_url;
+    // Ref here is the branch name
+    this.headRef = pr.head.ref;
+    this.headSha = pr.head.sha;
```

Comment:
I'm thinking aloud here: Does this need to account for the fact that a PR could have been created based on a commit that's older than the current `HEAD` on `ampproject/amphtml:master`? Also, what about PRs that originate from the `master` branch of a fork? (This is rare, but still possible.)

Reply:
very good questions. so the only usage here for sha and the branch ref (name) is to update the check-run, basically the sha and branch ref are sort of ids to the specific check-run, so i don't believe it matters if the commit is old or if the PR is from a branch or a fork. The evaluation and generation of the OWNERS data structure is always based off of the ampproject/amphtml:master. let me know if that makes sense, we can discuss in our 1 on 1 if you need further clarification (i'm also hazy on this with my on and off work on this).

Reply:
I think we're good here for now. Let's merge / deploy this and test in various ways.